### DISCUSSION TITLE
Modernizing tree-sitter on Atom

### DISCUSSION THREAD

**Discussion by mauricioszabo** [tree-sitter/tree-sitter#2044] [Q&A]
Hi - I'm one of the maintainers of Pulsar, an Atom fork, and one of the things I really want is to remake the implementation of tree-sitter there to use queries and to be compatible with the newer Electron. I know this question is not _really related_ to tree-sitter, but maybe someone is familiar with Atom and can help me...

So, for now I was able to make the queries work and I'm using the WASM based version. The problem is understanding how Atom tokenizer works - I tried to make a "language mode" for this newer implementation, but it seems I need to implement a `HighlightIterator` that I actually have no idea how it works - can someone point me to the right direction?

**Comment by sogaiu** [comment]
I think may be @maxbrunsfeld might know :)
